airbyte-integrations/connectors/source-shortio/source_shortio/source.py

```diff
@@ -0,0 +1,266 @@
+        if current_stream_state is not None and 'dt' in current_stream_state:
+            return {'dt' : self.before_dt}
+        else:
+            return {'dt' : self.config['start_date']}
+        
+    def request_body_json(
+        self, 
+        stream_state: Mapping[str, Any], 
+        stream_slice: Mapping[str, any] = None, 
+        next_page_token: Mapping[str, Any] = None
+    ) -> MutableMapping[str, Any]:
+        """
+        This method passes the arguments necessary to get the clicks from the shortio API. 
+        No parameters have been implemented here at all with the exception of hardcoding human clicks only to come through.
+        Human clicks are hardcoded to reduce unnecessary clicks from coming through.
+
+        :return dict: json body for the request
+        """
+        payload =  {
+            "limit" : self.limit,
+            "include" : {"human":True},
+            "beforeDate" :  next_page_token or self.before_dt,
+        }
+
+        if stream_state and 'dt' in stream_state.keys():
+            payload['afterDate'] = stream_state['dt']
+        else:
+            payload['afterDate'] = self.config['start_date']
+
+        # Capturing request
+        self.logger.log(level='INFO', message=str(payload))
+        return payload
+
+    def parse_response(self, response: requests.Response, **kwargs) -> Iterable[Mapping]:
+        yield from json.loads(response.text)
+
+
+# Source
+class SourceShortio(AbstractSource):
+    def check_connection(self, logger, config) -> Tuple[bool, any]:
+        """
+        CHeck whether configuration is correct. 
+
+        :param config:  the user-input config object conforming to the connector's spec.json
+        :param logger:  logger object
+        :return Tuple[bool, any]: (True, None) if the input config can be used to connect to the API successfully, (False, error) otherwise.
+        """
+        try:
+
+            url = "https://api.short.io/api/domains"
+            api_secret = config["secret_key"]        
+            domain_id = int(config["domain_id"])
+            headers = {
+                "Accept": "application/json",
+                "Authorization": api_secret
+            }
+
+            response = requests.request("GET", url, headers=headers)
+            response.raise_for_status()
+            for domain in response.json():
+                if domain_id == domain['id']:
+                    return True, None
+        except Exception as e:
+            return False, e
+
+        return False, 'Domain not found'
+
+    def streams(self, config: Mapping[str, Any]) -> List[Stream]:
+        """
+        :param config: A Mapping of the user input configuration as defined in the connector spec.
+        """
+        key = config['secret_key']
+        auth = BasicAuthenticator(token=key, auth_method=None)
+        links = Links(authenticator=auth)
+        links._config = config
```

Comment:
instead of passing the entire config here, let's actually have each field as private class vars in each stream, and pass the fields separately. It'll make it easier to tell what field is being used where. Does that make sense?